### PR TITLE
zerotier-one: service add

### PR DESCRIPTION
**Pull request by sjmackenzie**

### CODE REVIEW

**Comment by sjmackenzie** [comment]
@peti @7c6f434c :-) thanks

File: nixos/modules/services/networking/zerotierone.nix [hunk 1]
```diff
@@ -0,0 +1,41 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.zerotierone;
+in
+{
+  options.services.zerotierone = {
+    enable = mkOption {
```
Comment (jagajaga) [inline nixos/modules/services/networking/zerotierone.nix]: Use `mkEnableOption`.

File: nixos/modules/services/networking/zerotierone.nix [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+      preStart =
+        ''
+        mkdir -p /var/lib/zerotier-one
+        chmod 700 /var/lib/zerotier-one
+        chown -R root:root /var/lib/zerotier-one
+        '';
+      serviceConfig = {
+        Type = "forking";
+	User = "root";
+	PIDFile = "/var/lib/zerotier-one/zerotier-one.pid";
```
Comment (jagajaga) [inline nixos/modules/services/networking/zerotierone.nix]: Fix indentation please.

**Comment by jagajaga** [comment]
Don't forget to add `:` in the commits name :) `zerotier-one: service add`

**Comment by sjmackenzie** [comment]
@jagajaga

File: nixos/modules/services/networking/zerotierone.nix [hunk 3]
```diff
@@ -0,0 +1,35 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.zerotierone;
+in
+{
+  options.services.zerotierone.enable = mkEnableOption "Whether to enable ZeroTierOne, the flat virtual Ethernet networks of almost unlimited size.";
```
Comment (jagajaga) [inline nixos/modules/services/networking/zerotierone.nix]: You have to pass only service name to this function. Not a full sentence.

File: nixos/modules/services/networking/zerotierone.nix [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+        User = "root";
+        PIDFile = "/var/lib/zerotier-one/zerotier-one.pid";
+        ExecStart = "${pkgs.zerotierone}/bin/zerotier-one -d";
+      };
+    };
+  environment.systemPackages = [ pkgs.zerotierone ];
+  };
+}
+
+
```
Comment (jagajaga) [inline nixos/modules/services/networking/zerotierone.nix]: Remove these extra lines.

**Comment by sjmackenzie** [comment]
@jagajaga

File: nixos/modules/services/networking/zerotierone.nix [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+      wantedBy = [ "multi-user.target" ];
+      preStart =
+        ''
+        mkdir -p /var/lib/zerotier-one
+        chmod 700 /var/lib/zerotier-one
+        chown -R root:root /var/lib/zerotier-one
+        '';
+      serviceConfig = {
+        Type = "forking";
+        User = "root";
```
Comment (jagajaga) [inline nixos/modules/services/networking/zerotierone.nix]: Hmm. I'm not sure if it's ok to run as root.. 
Or this service needs root's rights?

Comment (sjmackenzie) [inline nixos/modules/services/networking/zerotierone.nix]: I'm afraid it needs root.

File: nixos/modules/services/networking/zerotierone.nix [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+  config = mkIf cfg.enable {
+    systemd.services.zerotierone = {
+      description = "ZeroTierOne";
+      path = [ pkgs.zerotierone ];
+      after = [ "network.target" ];
+      wantedBy = [ "multi-user.target" ];
+      preStart =
+        ''
+        mkdir -p /var/lib/zerotier-one
+        chmod 700 /var/lib/zerotier-one
```
Comment (dotlambda) [inline nixos/modules/services/networking/zerotierone.nix:20]: Why not `-R`?

**Pull request merged**